types/block.go

```diff
@@ -1261,6 +1262,9 @@ func (data *Data) ToProto() tmproto.Data {
 
 	protoMsgs := make([]*tmproto.Message, len(data.Messages.MessagesList))
 	for i, msg := range data.Messages.MessagesList {
+		if err := msg.ValidateBasic(); err != nil {
+			panic(err)
```

Comment:
we'll definitely have to bubble up this error eventually, which will actually be a fair amount of work iirc

Reply:
Bubbling as in changing the signature of the function and adding an error?

Reply:
yes, unfortunately. It's a fair amount of work :grimacing: as normie tendermint assumes this is a simply failproof function call in a lot of places

Reply:
it might be easier just to check these things in Prepare/ProcessPropsosal manually, and that will also allow us to keep a smaller diff with upstream

Reply:
actually, the more I think about it, the more I'm in favor of that solution, as we'd have to properly handle all of those errors here (with no context of the app) instead of in the application. That solution also has the benefit of keeping all of the block validity logic in the application, which is something that we definitely want to do.

Reply:
Wanted to add it there but then I found that in prepare/process we're handling the blocks and more general concepts. So reducing the granularity to messages would make the function look inconsistent. What do you think?

Reply:
Cool, will try to bubble the error then.

Reply:
> Wanted to add it there but then I found that in prepare/process we're handling the blocks and more general concepts. So reducing the granularity to messages would make the function look inconsistent. What do you think?
mind elaborating on this a bit more, I'm not sure what you mean yet. We can have a quick sync whenever you have time too

> will try to bubble the error then.

as mentioned briefly, I'm not sure this is the correct approach, as we're adding application logic (which namespaces are valid) to tendermint. I think the simpler approach of simply checking each namespace in both prepare/process proposal is more desirable because it keeps application logic in the app, and allows us to filter txs in prepareProposal instead of failing to produce a block

Reply:
> Wanted to add it there but then I found that in prepare/process we're handling the blocks and more general concepts. So reducing the granularity to messages would make the function look inconsistent. What do you think?
Never mind this. I didn't know we also have prepare/process implementation in the app. I was seeing the ones we have on execution, and they seemed rather general.